src/compiler/checker.ts

```diff
@@ -6662,46 +6677,95 @@ namespace ts {
             return s.valueDeclaration ? s.valueDeclaration.kind : SyntaxKind.PropertyDeclaration;
         }
 
-        function getDeclarationFlagsFromSymbol(s: Symbol) {
+        function getDeclarationFlagsFromSymbol(s: Symbol): NodeFlags {
             return s.valueDeclaration ? getCombinedNodeFlags(s.valueDeclaration) : s.flags & SymbolFlags.Prototype ? NodeFlags.Public | NodeFlags.Static : 0;
         }
 
-        function checkClassPropertyAccess(node: PropertyAccessExpression | QualifiedName, left: Expression | QualifiedName, type: Type, prop: Symbol) {
+        /**
+         * Check whether the requested property access is valid.
+         * Returns true if node passed the check, and false otherwise.
+         * @param node The node to be checked.
+         * @param left The left hand side of the property access (eg: the super is `super.*`).
+         * @param type The type of left.
+         * @param prop The symbol for the right hand side of the property access.
+         */
+        function checkClassPropertyAccess(node: PropertyAccessExpression | QualifiedName, left: Expression | QualifiedName, type: Type, prop: Symbol) : boolean {
             let flags = getDeclarationFlagsFromSymbol(prop);
-            // Public properties are always accessible
+            let declaringClass : InterfaceType;
+
+            if (left.kind === SyntaxKind.SuperKeyword) {
+                let errorNode = (<PropertyAccessExpression>node).name ?
+                    (<PropertyAccessExpression>node).name :
+                    (<QualifiedName>node).right;
+
+                // TS 1.0 spec (April 2014): 4.8.2
+                // - In a constructor, instance member function, instance member accessor, or
+                //   instance member variable initializer where this references a derived class instance,
+                //   a super property access is permitted and must specify a public instance member function of the base class.
+                // - In a static member function or static member accessor
+                //   where this references the constructor function object of a derived class,
+                //   a super property access is permitted and must specify a public static member function of the base class.
+                if (getDeclarationKindFromSymbol(prop) !== SyntaxKind.MethodDeclaration) { // prop is a property access

```

Comment:
It clarifies what the next test is doing to conform with the spec. I refactored this block of code from its previous location in one of the two callers because the other caller had the same logic. 

I actually think the comment clarifies why we need to check if the property accessed is a member function or not -- it's a useful form of documentation. 

Perhaps it's a bit verbose, however.